src/mlpack/methods/preprocess/preprocess_scale_main.cpp

```diff
@@ -0,0 +1,174 @@
+#include <mlpack/core/data/scaler_methods/standard_scaler.hpp>
+
+PROGRAM_INFO("Scale Data",
+    // Short description.
+    "A utility to perform feature scaling.",
+    // Long description.
+    "This utility takes a dataset and performs feature scaling using one of "
+    "the five scaler methods namely: max_abs_scaler, mean_normalization, "
+    "min_max_scaler ,standard_scaler and whitening. The function takes a "
+    "matrice as " + PRINT_PARAM_STRING("input") + " and a scaling method type"
+    " which you can specify using " + PRINT_PARAM_STRING("scaler_method") +
+    " parameter; the default is standard scaler, and outputs a matrice "
+    "with scaled feature."
+    "\n\n"
+    "The output scaled feature matrices may be saved with the " +
+    PRINT_PARAM_STRING("output") + " output parameters."
+    "\n\n"
+    "So, a simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled")+ " with "
+    " standard_scaler as scaler_method, we coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "standard_scaler") +
+    "\n\n"
+    "A simple example where we want to whiten the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_whitened")+ " with "
+    " PCA as whitening_method and use 0.01 as regularization parameter, "
+    "we coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "whitening", "whitening_method", "PCA", "epsilon", 0.01) +
+    "\n\n"
+    "Another simple example where we want to scale the dataset " +
+    PRINT_DATASET("X") + " into " + PRINT_DATASET("X_scaled") + " with "
+    " min_max_scaler as scaler method, where scaling range is 1 to 3 instead"
+    " of deafult 0 to 1. We coud run "
+    "\n\n" +
+    PRINT_CALL("preprocess_scale", "input", "X", "output", "X_scaled",
+    "scaler_method", "min_max_scaler", "min_value", 1, "max_value", 3),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// Define parameters for data.
+PARAM_MATRIX_IN_REQ("input", "Matrix containing data.", "i");
+PARAM_MATRIX_OUT("output", "Matrix to save scaled data to.", "o");
+PARAM_STRING_IN("scaler_method", "method to use for scaling, the "
+    "default is standard_scaler.", "a", "standard_scaler");
+PARAM_STRING_IN("whitening_method", "method to use for whitening, the "
+    "default is ZCA.", "w", "ZCA");
+PARAM_DOUBLE_IN("epsilon", "regularization Parameter, should be between"
+  "-1 to 1.", "r", 0.000001);
+
+PARAM_INT_IN("seed", "Random seed (0 for std::time(NULL)).", "s", 0);
+PARAM_INT_IN("min_value", "Starting value of range for min_max_scaler.",
+    "b", 0);
+PARAM_INT_IN("max_value", "Ending value of range for min_max_scaler.",
+    "e", 1);
+
```

Comment:
Another set of parameters you might consider including is a `PARAM_MODEL_IN()` and `PARAM_MODEL_OUT()` that allows saving of the scalers for reuse, i.e., for inverse scaling (to get back to the original matrix), or for scaling new data with the parameters of the old scaling.  However, I realize that would add a lot of complexity to this, and I'm fine with it as is if you don't want to add that now.

Reply:
Umm, I seriously have no clue how both of these work, and thus would like to study it and then if feasible would implement it, but i would keep that for later. I would learn about `PARAM_MODEL_IN()` after gsoc most probably.

As of now, I would like not to code it since i have no prior knowledge about it.

Reply:
> Another set of parameters you might consider including is a `PARAM_MODEL_IN()` and `PARAM_MODEL_OUT()` that allows saving of the scalers for reuse, i.e., for inverse scaling (to get back to the original matrix), or for scaling new data with the parameters of the old scaling. However, I realize that would add a lot of complexity to this, and I'm fine with it as is if you don't want to add that now.

ummm, so the PR didn't had any reviews and thus took some time out and implemented, `PARAM_MODEL_IN()` and `PARAM_MODEL_OUT()`, I changed the API a lilttle bit so that both these could be introduced, A function `Fit()` is introduced which could gain information about all the scaling and then the object can be used to transform multiple dataset with same scales and may not be required to calculate the scale again and again